### PR TITLE
Qpt vertica changes

### PR DESCRIPTION
**Pull request by Trianz-Akshay**
*Issue #, if available:*

*Description of changes:*
Query passthrough changes for vertica connector

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-vertica/pom.xml [hunk 1]
```diff
@@ -66,6 +66,12 @@
             <artifactId>ST4</artifactId>
             <version>${antlr.st4.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>athena-jdbc</artifactId>
+            <version>2022.47.1</version>
```
Comment (AbdulR3hman) [inline athena-vertica/pom.xml:72]: please don't explicitly chose the version. It should be done by maven.

File: athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java [hunk 2]
```diff
@@ -174,6 +203,79 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
...
+    {
+        ImmutableMap.Builder<String, List<OptimizationSubType>> capabilities = ImmutableMap.builder();
+        queryPassthrough.addQueryPassthroughCapabilityIfEnabled(capabilities, configOptions);
+
+        return new GetDataSourceCapabilitiesResponse(request.getCatalogName(), capabilities.build());
+    }
+
+
+    @Override
+    public GetTableResponse doGetQueryPassthroughSchema(final BlockAllocator blockAllocator, final GetTableRequest getTableRequest)
```
Comment (AbdulR3hman) [inline athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java:222]: The implementation of this method brings up few issues;

1. We are using JDBC Implementation here; (almost copy/paste from the JDBC Metadatahandler) whereas `doGetTable` uses `VerticaSchemaUtils` class. 
2. This brings up an important question; why are we not using the JDBC metadatahandler to begin with? If there is a good reason for it, then we shouldn't be using the JDBC implementation here either.
3. If it seems that there in fact we don't need to have different implementation between JDBC metadata handler and Vertica's, we need to use this opportunity to unify the code.

**Review by AbdulR3hman — DISMISSED** [review]
please look over the comments and let me know.

File: athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java (hunk 2 above)
Comment (Trianz-Akshay) [reply athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java:221]: incorporated the changes for vertica datatype to arrow in dogetQuerypassthroughSchema.

File: athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java [hunk 3]
```diff
@@ -243,20 +316,29 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request
...
         VerticaExportQueryBuilder queryBuilder = queryFactory.createVerticaExportQueryBuilder();
+        String preparedSQLStmt;
 
-        String preparedSQLStmt = queryBuilder.withS3ExportBucket(s3ExportBucket)
-                .withQueryID(queryID)
-                .withColumns(definition, schemaName)
-                .fromTable(tableName.getSchemaName(), tableName.getTableName())
-                .withConstraints(constraints, schemaName)
-                .build();
+        if (!request.getTableName().getQualifiedTableName().equalsIgnoreCase(queryPassthrough.getFunctionSignature())) {
```
Comment (aimethed) [inline athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java:326]: I think a better way to check if a request is a query passthrough request is using: `request.getConstraints().isQueryPassThrough()`

**Review by aimethed — COMMENTED** [review]
Some minor changes/suggestions I noticed.

File: athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java [hunk 4]
```diff
@@ -174,6 +202,51 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
...
+    @Override
+    public GetTableResponse doGetQueryPassthroughSchema(final BlockAllocator blockAllocator, final GetTableRequest getTableRequest)
+            throws Exception
+    {
+        if (!getTableRequest.isQueryPassthrough()) {
+            throw new IllegalArgumentException("No Query passed through [{}]" + getTableRequest);
+        }
+
+        queryPassthrough.verify(getTableRequest.getQueryPassthroughArguments());
+        String customerPassedQuery = getTableRequest.getQueryPassthroughArguments().get(JdbcQueryPassthrough.QUERY);
```
[the remark this answers is not in the document]
Comment (Trianz-Akshay) [reply athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java:229]: changed to VerticaQueryPassthrough.QUERY

**Review by aimethed — APPROVED** [review]

**Pull request merged**